beacon_node/beacon_chain/src/execution_payload.rs

```diff
@@ -426,10 +427,10 @@ pub fn get_execution_payload<
         &BeaconState::Base(_) | &BeaconState::Altair(_) => None,
     };
     let parent_beacon_block_root = match state {
-        &BeaconState::Deneb(_) => Some(state.latest_block_header().canonical_root()),
-        &BeaconState::Merge(_) | &BeaconState::Capella(_) => None,
+        BeaconState::Deneb(_) => Some(parent_block_root),
```

Comment:
I thought this was buggy too, because `latest_block_header` doesn't have the `state_root` filled in _in general_. However in this case it's actually fine, because the state has been advanced to the proposal slot. Still, it's slightly quicker to reuse the parent block hash rather than recomputing it, so I kept this change.

Reply:
Nice!